Aztec/Classes/NSAttributedString/Conversions/AttributedStringParser.swift

```diff
@@ -381,17 +384,15 @@ private extension AttributedStringParser {
     private func mergeablePair(from mergeableNodes: [MergeablePair]) -> MergeablePair? {
         assert(mergeableNodes.count > 0)
         
-        // Business logic: The last mergeable node is never merged, so we need more than 1 node to continue.
-        //
-        guard let lastNodeName = mergeableNodes.last?.left.name else {
+        guard let lastNode = mergeableNodes.last?.left else {
             return nil
         }
+        
+        let lastNodeName = lastNode.name
 
         var mergeCandidates: ArraySlice<MergeablePair>
         
-        // TODO: Remove this hack!!  This is a horrible horrible hack, but it's the simplest solution until we can
-        // refactor this Parser to work in a different way, analyzing the NSAttributedString directly for merges.
-        if mergeableNodes.last?.left.type == .figure {
+        if Element.mergeableBlockLevelElementWithoutBlockLevelChildren.contains(lastNode.type) {
```

Comment:
Should we keep the comment above or this is no longer an hack now ?

Reply:
The comment I put there wasn't clear even to me (and I wrote it).  It was offering no value, in my opinion.

Since it wasn't clear at all, I prefer the code to speak for itself, and let whoever analyzes it draw his/her own conclusions.

Reply:
PS: also, it doesn't really apply to the line I put it in... it's more of a comment on the whole class, but even if I add it back I'd have to think how to word it for clarity.